### PR TITLE
Make layout smaller screen size friendly

### LINKED ISSUE #11: Proper table column resize

### LINKED ISSUE #58: make different views for smaller screen friendly
restyle for smaller screens/view sizes

### PR DESCRIPTION
**Pull request by mahos**
Make column size fit either the header or the body content - whichever is larger and also make the columns resizable.
fixes #11 
resize column 1

Make layout change for smaller screens (still in works)
fixes #58
labbook resize

### CODE REVIEW

File: src/Components/MainTableView/TableContent.tsx [hunk 1]
```diff
@@ -79,6 +81,10 @@ export default class TableContent extends React.Component<TableContentProps, Tab
     this.goForwardAPage = this.goForwardAPage.bind(this);
     this.goBackwardAPage = this.goBackwardAPage.bind(this);
     this.handleNumberOfTuplesPerPageChange = this.handleNumberOfTuplesPerPageChange.bind(this);
+  
+    this.headerColumnSizeRef = React.createRef();
+    this.tableBodyColumnRefs = [];
+    // this.bodyColumnSizeRef = React.createRef();
```
Comment (guzman-raphael) [inline src/Components/MainTableView/TableContent.tsx:87]: Lingering debug comment.

File: src/Components/MainTableView/TableContent.tsx [hunk 2]
```diff
@@ -97,6 +103,57 @@ export default class TableContent extends React.Component<TableContentProps, Tab
     this.setState({currentSelectedTableActionMenu: TableActionType.FILTER, hideTableActionMenu: true, selectedTuple: undefined});
   }
 
+  /**
+   * initialzing the table column width 
+   */
+  componentDidMount() {
+    let tableBodyCellWidthLookup: Array<Array<number>> = []
+    let tablenewHeaderWidths: Array<number> = []
+    // let headerColumns = this.headerColumnSizeRef.current.cells
```
Comment (guzman-raphael) [inline src/Components/MainTableView/TableContent.tsx:112]: Lingering debug comment.

File: src/Components/MainTableView/TableContent.tsx [hunk 3]
```diff
@@ -97,6 +103,57 @@ export default class TableContent extends React.Component<TableContentProps, Tab
...
+    }
+
+    // going through the body column references to decide on the initial width of the table columns
+    let finalTableColWidths: Array<number> = []
+    tableBodyCellWidthLookup.forEach((col: Array<number>, index: number) => {
+      if (col.length > 0) {
+        let colAvg = Math.ceil(col.reduce((a: number, b: number) => (a + b)) / col.length)
+        //// checking the max width of each column just in case - implementing with rounded up average for now
+        // let colMax = Math.max(...col)
+        // console.log(`Avg: ${colAvg}...Max: ${colMax}`)
```
Comment (guzman-raphael) [inline src/Components/MainTableView/TableContent.tsx:142]: Lingering debug comments.

**Review by guzman-raphael — CHANGES_REQUESTED** [review]

**Comment by guzman-raphael** [comment]
@mahos Would you also assign this PR to a Milestone?

File: src/Components/MainTableView/TableContent.tsx (hunk 1 above)
Comment (mahos) [reply src/Components/MainTableView/TableContent.tsx:87]: addressed

File: src/Components/MainTableView/TableContent.tsx (hunk 2 above)
Comment (mahos) [reply src/Components/MainTableView/TableContent.tsx:112]: addressed

File: src/Components/MainTableView/TableContent.tsx (hunk 3 above)
Comment (mahos) [reply src/Components/MainTableView/TableContent.tsx:142]: addressed

**Review by guzman-raphael — DISMISSED** [review]
@mahos Good from my side. Will give @Synicix an opportunity to give a last look and go ahead with merge.

**Review by Synicix — CHANGES_REQUESTED** [review]
Just one syntax change

File: src/Components/MainTableView/TableContent.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
-import React from 'react';
+import React, { RefObject } from 'react';
```
[the remark this answers is not in the document]
Comment (mahos) [reply src/Components/MainTableView/TableContent.tsx]: yes - good catch - it's the auto import that puts those spaces by default...-_-

**Review by Synicix — APPROVED** [review]

**Pull request merged**